test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTypesTest.java

```diff
@@ -0,0 +1,1215 @@
+     */
+    @Test
+    public void testFrozenUDT()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a frozen<udt>, b frozen<udt>)")
+                .populateNode("INSERT INTO %s (k, a, b) VALUES (0, {x:1, y:2}, {x:3, y:4})")
+                .queryColumns("a.x",
+                              rows(row(1)),
+                              rows(row(0, udt("x", 1, "y", 2), udt("x", 3, "y", 4))),
+                              rows(row(0, udt("x", 1, "y", 2), null)))
+                .deleteColumn("DELETE a FROM %s WHERE k=0",
+                              "b.y",
+                              rows(row(4)),
+                              rows(row(0, null, udt("x", 3, "y", 4))),
+                              rows(row(0, udt("x", 1, "y", 2), udt("x", 3, "y", 4))))
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, null, udt("x", 3, "y", 4))))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen sets.
+     */
+    @Test
+    public void testNonFrozenSet()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a set<int>, b set<int>, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, {1, 2}, {3, 4}, 10)")
+                .queryColumns("a[1]",
+                              rows(row(1)),
+                              rows(row(0, set(1, 2), set(3, 4), 10)),
+                              rows(row(0, set(1), set(3, 4), null))) // unselected 'b' is repaired, see CASSANDRA-10657
+                .deleteColumn("UPDATE %s SET a=a-{2} WHERE k=0",
+                              "b[4]",
+                              rows(row(4)),
+                              rows(row(0, set(1), set(3, 4), 10)),
+                              rows(row(0, set(1), set(3, 4), 10))) // unselected 'a' is repaired, see CASSANDRA-10657
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, set(1), set(3, 4), 10)))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen lists.
+     */
+    @Test
+    public void testNonFrozenList()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a list<int>, b list<int>, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, [1, 2], [3, 4], 10)")
+                .queryColumns("a",
+                              rows(row(list(1, 2))),
+                              rows(row(0, list(1, 2), list(3, 4), 10)),
+                              rows(flush ? row(0, list(1, 2), list(null, null), null)
+                                         : row(0, list(1, 2), list(3, 4), null))) // TODO: is this RR propagation correct?
```

Comment:
I want to say it isn't...like we shouldn't touch `b` at all for a RR that only selects `a`?

Reply:
CC @beobal